api/docs/reference/search_algorithm.md

```diff
@@ -0,0 +1,226 @@
+# Search Algorithm
+
+Openverse currently uses a relatively simple and naïve search algorithm with
+very limited options. The documentation on this page was written by referencing
+the code in Openverse as well as parts of Openverse's historical development.
+Parts of the story for how Openverse's indexes came to be configured as they are
+today are likely missing. Future improvements to Openverse's indexing and search
+will be more carefully documented here and in the code to ensure there is
+greater longevitiy of understanding.
+
+> **Note**: This document avoids covering details covered in the
+> [Openverse Search Guide](https://wordpress.org/openverse/search-help).
+> Specifically, this document does not describe _how_ to search (advanced
+> techniques and syntax), rather _what is searched_ and in what way.
+
+## Reference resources
+
+This document includes links to specific parts of Elasticsearch's documentation
+and Openverse code. The following are broadly useful entry points into learning
+about Elasticsearch, full text-search, and Openverse's index configuration:
+
+- [Elasticsearch 7.12 documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/index.html)
+- [Full-text search (Wikipedia)](https://en.wikipedia.org/wiki/Full-text_search)
+- [Stemming (Wikipedia)](https://en.wikipedia.org/wiki/Stemming)
+- [`es_mapping.py` (index configuration)](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/ingestion_server/es_mapping.py)
+
+## Terms
+
+- "Document": The total queryable information representing a single work
+  catalogued in Openverse.
+- "Field": The individual queryable elements of a document. In the context of
+  Openverse, these may be textual, numerical, or keywords.
+- "Result relevance": An individual result is "relevant" to a search when it
+  matches the expectations of the user for a given query. A result is irrelevant
+  if it does not match the expectations of the user for that same query. For
+  example, the query "bird watch" may produce pictures of a wrist watch with a
+  bird clockface illustration _or_ could surface pictures related to the
+  activity also known as "birding" (bird watching), due to stemming. In the case
+  of this specific query, "bird watching" may not be relevant, despite being a
+  technically correct match for the query given Openverse's current index
+  configuration. Other relevancy issues may be caused by descriptions that are
+  not related to the contents of an image. This often happens on Flickr where
+  users sometimes include blog-like text in the description of an image that
+  references things that happened outside of the context of the image itself.
+- "Result quality": A combination of relevance and other factors like the actual
```

Comment:
Nice distinction! I haven't thought about result quality being different from the result relevance before.

Reply:
I had been tinkering with the idea in my head a bit but only when writing this document did I realise how important a distinction it actually is. We'll need to keep it in mind as we dive into improving search.